### PR TITLE
fix(animation): not defaulting starting value from animated value

### LINKED ISSUE #11896: Animation's `ComputeFromValue` is not working correctly
### Current behavior

The `Expander` style in WinUI has the following in `CheckedPointerOver` `VisualState`:

```xaml
  <contract7NotPresent:DoubleAnimation
      Storyboard.TargetName="ExpandCollapseChevronRotateTransform"
      Storyboard.TargetProperty="Angle"
      To="180.0" Duration="0:0:0.1"/>
```

However, the same is in the `Checked` `VisualState` to accomodate for keyboard navigation. When the animation finishes, the final value is kept, so subsequent pointer hover should not cause further animation (as the effective `From` value is already 180.0). However, in case of Uno, the chevron snaps to angle 0 and rotates quickly. 

### Expected behavior

No animation should occur on poitner over.

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

Our current workaround in `Expander_themeresources.xaml` was to replace all ocurrences of `Duration="0:0:0.1"` by `Duration="0:0:0"`. Revert this workaround to see the broken behavior.

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia, Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

Visual Studio 2022

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Xiaoy312**
GitHub Issue (If applicable): 
- closes unoplatform/nventive-private#464, 
- closes unoplatform/nventive-private#452
- closes #11896

## PR Type
- Bugfix

## What is the current behavior?
Currently animations with omitted `From` infers their starting value from the non-animated value.
On windows, you are allow to starting a new animation from the animated value, be mid animation or the `HoldEnd` value of a completed animation.

## What is the new behavior?
Aligned behavior with windows:
animation start from animated value

## PR Checklist
Please check if your PR fulfills the following requirements:
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

**Comment by Xiaoy312** [comment]
reserving top comment

File: src/Uno.UI/UI/Xaml/Media/Animation/Animators/AnimatorFactory.Android.cs [hunk 1]
```diff
@@ -332,10 +332,57 @@ private static ValueAnimator GetRelativeAnimator(Java.Lang.Object target, string
...
+			if (timeline.GetIsDependantAnimation() || timeline.GetIsDurationZero())
+			{
+				return false;
+			}
+
+			var info = timeline.PropertyInfo.GetPathItems().Last();
+			var target = info.DataContext;
+			var property = info.PropertyName
+				.Split(new[] { '.' }).Last()
+				.Replace("(", "").Replace(")", "");
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Media/Animation/Animators/AnimatorFactory.Android.cs:353]: There's already a parser for that if I'm not mistaken, we probably should be reusing it.

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_DoubleAnimation.cs [hunk 2]
```diff
@@ -162,5 +165,68 @@ public async void When_RepeatForever_ShouldLoop_AsdAsd()
 			Assert.AreEqual(10d, averageIncrement, 1.5, "an rough average of increment (exluding the drop) of 10 (+-15% error margin)");
 			Assert.IsTrue(incrementSizes.Count(x => x > 3) > 8, $"at least 10 (-2 error margin: might miss first and/or last) sets of continuous increments that size of 4 (+-1 error margin: sliding slot): {string.Join(",", incrementSizes)}");
 		}
+
+		[TestMethod]
+		[RunsOnUIThread]
+#if __ANDROID__
+		[Ignore("The native property (translate.View.Y) is being animated here instead of the dependency property (translate.Y).")]
```
Comment (dr1rrb) [inline src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_DoubleAnimation.cs:172]: It seems that you added some code specific to Android to handle that case, is this `[Ignore]` still relevant ? If yes, could we add another test that will validate that platform specific code?

File: src/Uno.UI/UI/Xaml/Media/Animation/Animators/AnimatorFactory.Android.cs (hunk 1 above)
Comment (Xiaoy312) [reply src/Uno.UI/UI/Xaml/Media/Animation/Animators/AnimatorFactory.Android.cs]: Didn't find any, so I added one.

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_DoubleAnimation.cs [hunk 3]
```diff
@@ -116,37 +120,42 @@ async Task WaitAndSnapshotValue(int millisecondsDelay)
...
 		[RunsOnUIThread]
-		public async void When_RepeatForever_ShouldLoop_AsdAsd()
+		public async Task When_RepeatForever_ShouldLoop()
 		{
+			const int TimeResolutionScaling =
+#if !DEBUG && (__ANDROID__ || __MACOS__)
+				5;
+#else
+				1;
+#endif
```
Comment (jeromelaban) [inline src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_DoubleAnimation.cs:130]: Can you add a comment to explain why this is needed?

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_DoubleAnimation.cs [hunk 4]
```diff
@@ -116,37 +120,42 @@ async Task WaitAndSnapshotValue(int millisecondsDelay)
 
 		[TestMethod]
 		[RunsOnUIThread]
-		public async void When_RepeatForever_ShouldLoop_AsdAsd()
+		public async Task When_RepeatForever_ShouldLoop()
```
Comment (jeromelaban) [inline src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_DoubleAnimation.cs:123]: What the existing test failing ? If if did not, let's create another one.

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_DoubleAnimation.cs [hunk 5]
```diff
@@ -157,10 +166,73 @@ public async void When_RepeatForever_ShouldLoop_AsdAsd()
...
+			{
+				Assert.Fail($"Translate.Y is not a double: value={value}");
+			}
+
+			object GetTargetValue() =>
+#if !HAS_UNO
+				translate.GetValue(TranslateTransform.YProperty);
+#elif __ANDROID__
+				(double)translate.View.TranslationY;
+#else // uno, except android
```
Comment (jeromelaban) [inline src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_DoubleAnimation.cs:233]: Can you add the reason for why android is different?

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_DoubleAnimation.cs (hunk 4 above)
Comment (Xiaoy312) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Animation/Given_DoubleAnimation.cs:123]: Answered above.

**Review by dr1rrb — APPROVED** [review]

**Comment by Guidemarcus** [comment]
@jeromelaban could this be backported to uno 4.8 please?

**Pull request merged**